plugin/storage/memory/factory_test.go

```diff
@@ -80,8 +79,7 @@ func TestPublishOpts(t *testing.T) {
 	defer baseMetrics.Stop()
 	forkFactory := metricstest.NewFactory(time.Second)
 	defer forkFactory.Stop()
-	metricsFactory := fork.New("internal", forkFactory, baseMetrics)
-	require.NoError(t, f.Initialize(metricsFactory, zap.NewNop()))
+	require.NoError(t, f.Initialize(forkFactory, zap.NewNop()))
 
 	forkFactory.AssertGaugeMetrics(t, metricstest.ExpectedMetric{
```

Comment:
you changed PublishOpt to use expvar, metric factories are no longer relevant. You need to validate that the correct expvar is set (you can see how to do that in expvar/factory_test.go)

Reply:
thanks for the helps